### DISCUSSION TITLE
Regarding SHAP Values obtained from different models

### DISCUSSION THREAD

**Discussion by gaur41** [shap/shap#2742] [Q&A]
Hi Scott,

I hope you're doing well. I am applying SHAP analysis on different ML models like random forests, XGBoost, neural networks, and Support vector regression. However, I am getting slightly different summary pots for all methods (i.e., distribution of individual feature very over different models). How can I explain it as I am using the same input data in all four models?

Thanks

Have a good day!

**Comment by jainraj** [comment]
My two cents: Even if you are using same input data - the models are different! They learn different patterns from the data. SHAP helps you explain what the model has learnt from the data. Hence, different plots actually makes sense. 

The general idea is to build a model which is accurate. We then assume that this model has captured the actual variations in the data properly and try to explain this model - and thus helping us understand how the input and output are actually interacting. We use SHAP to confirm if the model has learnt right things or relied on some spurious patterns!
